File: apps/web/src/types/content.ts
export type Theme = {
  slug: string;
  name: string;
  description: string;
  articleCount: number;
};

export type ArticleCardItem = {
  id: string;
  themeSlug: string;
  title: string;
  summary: string;
  sourceUrl: string;
  publishedAt: string;
  stanceLabel: string;
  pointCount: number;
};

export type ThemeDetail = Theme & {
  articles: ArticleCardItem[];
};

export type ArticleDetail = {
  id: string;
  themeSlug: string;
  title: string;
  sourceUrl: string;
  sourceSiteLabel: string;
  publishedAt: string;
  translation: string;
  summary: string;
  keyPoints: string[];
  stanceLabel: string;
};

export type AdminJob = {
  id: string;
  type: "ingest" | "resummarize";
  status: "queued" | "running" | "completed" | "failed";
  targetLabel: string;
  requestedAt: string;
};

export type ContentApiMode = "mock" | "live";

export type AdminArticleOption = {
  id: string;
  title: string;
  themeSlug: string;
};

export type AdminArticleOptionsByTheme = Record<string, AdminArticleOption[]>;

export type QueueIngestionInput = {
  themeSlug: string;
  requestedBy?: string;
  idempotencyKey?: string;
};

export type QueueResummarizationInput = {
  articleId: string;
  requestedBy?: string;
  idempotencyKey?: string;
};

export type AdminActionResult = {
  ok: boolean;
  message: string;
};

File: apps/api/internal/domain/content.go
package domain

import "time"

// Theme は技術テーマを表すドメインモデル。
// テーマは Reddit 議論を分類する単位となり、複数の記事(Article)を持つ。
type Theme struct {
	Slug         string
	Name         string
	Description  string
	ArticleCount int
	UpdatedAt    time.Time
}

// ArticleCard は記事一覧で使用する軽量な記事表現。
// 詳細本文(KeyPointsの中身)を省略し、リスト表示に必要な情報のみを持つ。
type ArticleCard struct {
	ID          string
	ThemeSlug   string
	Title       string
	Summary     string
	SourceURL   string
	PublishedAt time.Time
	StanceLabel string
	PointCount  int
}

// ArticleDetail は記事詳細で使用する完全な記事表現。
// 翻訳文、要約、キーポイント全文を含む。
type ArticleDetail struct {
	ID              string
	ThemeSlug       string
	Title           string
	SourceURL       string
	SourceSiteLabel string
	PublishedAt     time.Time
	Translation     string
	Summary         string
	KeyPoints       []string
	StanceLabel     string
}
